tools/distrib/python/grpcio_tools/grpc_tools/test/protoc_test.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2020 The gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for protoc."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import contextlib
+import functools
+import multiprocessing
+import sys
+import unittest
+
+
+def _wrap_in_subprocess(error_queue, fn):
+
+    @functools.wraps(fn)
+    def _wrapped():
+        try:
+            fn()
+        except Exception as e:
+            error_queue.put(e)
+            raise
+
+    return _wrapped
+
+
+def _run_in_subprocess(test_case):
+    error_queue = multiprocessing.Queue()
+    wrapped_case = _wrap_in_subprocess(error_queue, test_case)
+    proc = multiprocessing.Process(target=wrapped_case)
+    proc.start()
+    proc.join()
+    if not error_queue.empty():
+        raise error_queue.get()
+    assert proc.exitcode == 0, "Process exited with code {}".format(
+        proc.exitcode)
```

Comment:
nit: This is a pretty good mechanism. Can we copy it to a common place and reuse between both tests?

Reply:
I thought about this, but realized it would be hard since we don't currently have a mechanism for sharing code between `grpcio_tools` and `grpcio_tests`. I've filed an issue to take care of this.